src/lib/ivy.ts

```diff
@@ -26,13 +26,15 @@ export interface PipeType<T> extends Type<T> {
 
 // As directive and component definitions are considered private API,
 // so those properties are prefixed with Angular's marker for "private".
+// An abstract component is not decorated by angular, so we should return the prototype.
```

Comment:
Abstract classes are decorated by decorators especially if they have declared bindings. Angular 9 introduced an empty Directive decorator that can be applied specifically for abstract classes.

Reply:
Okey, I will check that. The reproducible example has an Abstract Component with no decorators. We should think if the idea is make the decorator mandatory for abstract component or allow either

Reply:
Hey @arturovt I checked about the empty Directive decorator. Is mandatory when we want to use some Angular features in an Abstract Component. I updated the PR to support both, decorated and no decorated abstract components.